src/error.rs

```diff
@@ -0,0 +1,63 @@
+use std::fmt;
+use std::io;
+use std::error::Error as StdError;
+
+use futures::future::ExecuteError;
+use futures::sync::oneshot::Canceled;
+
+/// The Error type for the futures fs library
+#[derive(Debug)]
+pub enum Error {
```

Comment:
Using `oneshot::spawn_fn` will mean we don't need either canceled nor executor errors, so we can still just return `io::Error`.

Reply:
Based on what I've sent in the other thread, do you think it makes sense to unwrap or expect in order to clean the API and remove the error type?

Reply:
I think it probably makes sense to unwrap or expect, as the only errors that would occur is if the executor were falling apart, in which case things are probably pretty bad already.